### PR TITLE
Up to a working example!

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+pub mod verdict {
+    use libc::*;
+    use ffi::*;
+
+    pub enum Verdict {
+        Drop,
+        Accept,
+        Stolen,
+        Queue,
```
Comment (lclarkmichalek) [inline src/message.rs:41]: Queue should prob be Queue(u16) as the verdict's target queue is encoded in the top 16 bytes

File: src/message.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        Repeat,
+        Stop
+    }
+
+    pub fn set_verdict(qh: *mut nfq_q_handle, packet_id: u32, verdict: Verdict, data_len: u32, buffer: *const c_uchar) -> Result<c_int, ()> {
+        let c_verdict = match verdict {
+            Verdict::Drop => NF_DROP,
+            Verdict::Accept => NF_ACCEPT,
+            Verdict::Stolen => NF_STOLEN,
+            Verdict::Queue => NF_QUEUE,
```
Comment (lclarkmichalek) [inline src/message.rs]: So I have this as Verdict::Queue(n) => 3 | ((n as c_uint) << 16),

File: src/message.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            Verdict::Repeat => NF_REPEAT,
+            Verdict::Stop => NF_STOP
+        };
+
+        match unsafe { nfq_set_verdict(qh, packet_id as uint32_t, c_verdict as uint32_t, data_len as uint32_t, buffer) } {
+            -1 => Err(()),
```
Comment (lclarkmichalek) [inline src/message.rs]: Eww

File: src/ffi.rs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                                queuelen: uint32_t) -> c_int;
+
+    // Iterating through a queue
+    pub fn nfq_fd(handle: *mut nfq_handle) -> c_int;
+    pub fn nfq_handle_packet(handle: *mut nfq_handle,
+                             buf: *mut c_char,
+                             len: c_int) -> c_int;
+
+    // Deciding on a verdict
+    pub fn nfq_set_verdict(handle: *mut nfq_q_handle,
```
Comment (lclarkmichalek) [inline src/ffi.rs:70]: Will need nfq_set_verdict2 if support for setting the mark is wanted

File: src/handle.rs [hunk 5]
```diff
@@ -1,72 +1,106 @@
...
         } else {
             Ok(())
         }
     }
 
-    pub fn unbind(&mut self, proto: ProtoFamily) -> Result<(), NFQError> {
-        let _g = NFQ_LOCK.lock().unwrap();
+    /// Unbind the handle from a `Protocol Family`
+    ///
+    /// This should usually be avoided, as it may attach other programs from the `Protocol Family`.
```
Comment (lclarkmichalek) [inline src/handle.rs:69]: May unattach, surely?

**Comment by zzmp** [comment]
Ahh! I didn't even see these comments - I assumed once the merge was through, there was nothing else to say.

Anything you implemented since is still a part of my new PR. That doc comment is probably still wrong.
Can you explain what "setting the mark" means?

**Pull request merged**